flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java

```diff
@@ -166,7 +191,36 @@ public static MesosTaskManagerParameters create(Configuration flinkConfig) {
 			cpus,
 			containerType,
 			Option.apply(imageName),
-			containeredParameters);
+			containeredParameters,
+			constraints);
+		}
+
+	private static List<ConstraintEvaluator> parseConstraints(String mesosConstraints) {
+		List<ConstraintEvaluator> constraints = new ArrayList<>();
+
+		if (mesosConstraints != null) {
+			for (String constraint : Arrays.asList(mesosConstraints.split(","))) {
+				if (constraint.isEmpty()) {
+					continue;
+				}
+				final List<String> constraintList = Arrays.asList(constraint.split(":"));
+				if (constraintList.size() != 2) {
+					continue;
+				}
+				addConstraint(constraints, constraintList);
+			}
+		}
+
+		return constraints;
+	}
+
+	private static void addConstraint(List<ConstraintEvaluator> constraints, final List<String> constraintList) {
```

Comment:
Would you mind replacing `constraintList` with a pair of arguments for the attr name and value, and rename the `addConstraint` method to `addHostAttrValueConstraint`?